pkg/cmd/cli/cmd/deploy.go

```diff
@@ -225,11 +225,12 @@ func (o DeployOptions) deploy(config *deployapi.DeploymentConfig, out io.Writer)
 		}
 	}
 
-	config.Status.LatestVersion++
-	_, err = o.osClient.DeploymentConfigs(config.Namespace).Update(config)
+	config.Generation++
```

Comment:
Oh, and I didn't include 1) `ProgressDeadlineSeconds` which is still cooked and 2) the [rollback spec](https://github.com/kubernetes/kubernetes/blob/f29d597d0214758fd9f460a461f61624327e0103/pkg/apis/extensions/types.go#L257) but the latter should never be set by users anyway.